nodes/mtnode.py

```diff
@@ -71,17 +74,41 @@ def __init__(self):
 
         self.frame_local = get_param('~frame_local', 'ENU')
 
-        self.diag_msg = DiagnosticArray()
+
         self.stest_stat = DiagnosticStatus(name='mtnode: Self Test', level=1,
                                            message='No status information')
         self.xkf_stat = DiagnosticStatus(name='mtnode: XKF Valid', level=1,
                                          message='No status information')
         self.gps_stat = DiagnosticStatus(name='mtnode: GPS Fix', level=1,
                                          message='No status information')
-        self.diag_msg.status = [self.stest_stat, self.xkf_stat, self.gps_stat]
+
+	self.device_info = {
+		'product_code': self.mt.GetProductCode(),
+                'device_id': self.mt.GetDeviceID(),
+                'firmware_rev': self.mt.GetFirmwareRev()
+        }
+
+        output_config = self.mt.GetConfiguration()
+        for (k,v) in output_config.iteritems():
+            self.device_info[k] = v
+
+        self.updater = diagnostic_updater.Updater()
+        self.updater.setHardwareID(str(self.mt.GetDeviceID()))
+
+        self.updater.add("Self Test", self.diagnostic_self_test)
+        self.updater.add("XKF", self.diagnostic_xkf)
+        self.updater.add("GPS Fix", self.diagnostic_gps)
+        self.updater.add("Device Info", self.diagnostic_device)
+
+        self.imu_freq = diagnostic_updater.TopicDiagnostic("imu/data", self.updater,
+		diagnostic_updater.FrequencyStatusParam({'min': 0, 'max': 100}, 0.1, 10),
```

Comment:
I think it would make the most sense to check within a certain tolerance of the desired frequency, which is what I've done elsewhere (+/- 1% or +/- 5% depending on the hardware)